client/configuration/search_path.py

```diff
@@ -35,11 +44,11 @@ def _expand_relative_root(path: str, relative_root: str) -> str:
 
 class Element(abc.ABC):
     @abc.abstractmethod
-    def path(self) -> str:
+    def path(self) -> Union[str, None]:
```

Comment:
nit: Optional[str]

could you explain the reasoning and semantic differences for making this optional in the summary?